product_kits_inventory/product.py

```diff
@@ -0,0 +1,313 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (c) 2010-2014 Elico Corp. All Rights Reserved.
+#    Alex Duan <alex.duan@elico-corp.com>, Liu Lixia<liu.lixia@elico-corp.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp.osv import orm, fields, osv
+from openerp.tools.translate import _
+import openerp.addons.decimal_precision as dp
+
+
+class ProductProduct(orm.Model):
+    _inherit = 'product.product'
+
+    def _kits_product_available(self, cr, uid, ids, field_names=None,
+                                arg=False, context=None):
+        res = {}
+        field_names = field_names or []
+        context = context or {}
+        for id in ids:
+            res[id] = {}.fromkeys(field_names, 0.0)
+        field_map = {
+            'kits_qty_available': 'qty_available',
+            'kits_incoming_qty': 'incoming_qty',
+        }
+        for product_record in self.browse(cr, uid, ids, context=context):
+            # check if is a kit product.
+            so_qty = self._get_sale_quotation_qty(cr, uid, product_record.id,
+                                                  context=context)
+            if not self._is_kit(
+                cr, uid,
+                [product_record.id],
+                    context=context).get(product_record.id):
+
+                res[product_record.id] = {
+                    'kits_qty_available': 0,
+                    'kits_incoming_qty': 0,
+                    'kits_sale_quotation_qty': so_qty
```

Comment:
Why do you put a value on 'kits_sale_quotation_qty' if the product is not a kit ?